src/meta/src/barrier/mod.rs

```diff
@@ -325,13 +394,26 @@ enum BarrierEpochState {
     /// This barrier is current in-flight on the stream graph of compute nodes.
     InFlight,
 
-    /// This barrier is completed or failed.
-    Completed(Vec<BarrierCompleteResponse>),
+    /// This barrier is collected.
+    Collected(Vec<BarrierCompleteResponse>),
 }
 
-/// The result of barrier completion.
+enum CompletingCommand {
+    None,
+    Completing {
+        command_ctx: Arc<CommandContext>,
+
+        // The join handle of a spawned task that completes the barrier.
+        // The return value indicate whether there is some create streaming job command
+        // that has finished but not checkpointed. If there is any, we will force checkpoint on the next barrier
+        join_handle: JoinHandle<MetaResult<bool>>,
```

Comment:
Can we wrap the `require_checkpoint` flag in a struct, for easier extension in the future?